ipyparallel/controller/hub.py

```diff
@@ -833,15 +833,26 @@ def mia_task_request(self, idents, msg):
 
     #--------------------- IOPub Traffic ------------------------------
 
-    def save_iopub_message(self, topics, msg):
-        """save an iopub message into the db"""
-        # print (topics)
+    def monitor_iopub_message(self, topics, msg):
+        '''intercept iopub traffic so events can be acted upon'''
```

Comment:
Ok, good to know, I was torn.

I'm still not 100% certain that I'm cleaning everything up correctly: the only other path using 'unregister_engine' is when the heartbeat fails, which is similar to this case, but not exactly the same.  IE: I don't know what should happen to outstanding requests.  Any tips?